lib/experiment/factory.rb

```diff
@@ -1,15 +1,31 @@
 # Provides factory methods for storing singleton instance of Client
 module AmplitudeExperiment
-  @instances = {}
+  @local_instance = {}
+  @remote_instance = {}
   @default_instance = '$default_instance'
 
   # Initializes a singleton Client. This method returns a default singleton instance, subsequent calls to
   #  init will return the initial instance regardless of input.
   #
   # @param [String] api_key The environment API Key
   # @param [Config] config Optional Config.
-  def self.init(api_key, config = nil)
-    @instances.store(@default_instance, Client.new(api_key, config)) unless @instances.key?(@default_instance)
-    @instances.fetch(@default_instance)
+  def self.init_remote(api_key, config = nil)
```

Comment:
Now that we're not confined by reusing the `initialize` function, can we call these `initialize_remote` and `initialize_local` to keep consistency with other SDKs?